### PR TITLE
fix: Correct errors found while testing against the API

### PR DESCRIPTION
**Pull request by willsantanentcs**
While testing this code against the SmarterU API, I discovered that the "Email" and "EmployeeID" fields cannot be omitted when creating a User that does not have one of those attributes, and instead the tag should be included but left blank. I also discovered an error with the type casting in the getGroup() method. If approved, this PR will change the behavior of XMLGenerator to do this instead of omitting these fields entirely, and correctly cast the Group's notification emails to strings before adding them to the Group object.

### CODE REVIEW

File: src/XMLGenerator.php [hunk 1]
```diff
@@ -66,9 +66,13 @@ public function createUser(
         }
         if (!empty($user->getEmail())) {
             $info->addChild('Email', $user->getEmail());
+        } else {
+            $info->addChild('Email');
```
Comment (tomegantcs) [inline src/XMLGenerator.php:70]: I see that adding the empty `EmployeeID` node required a unit test change (as could be expected) however adding the empty `Email` node does not? Are we missing something in the unit tests?
